lib/IDE/Refactoring.cpp

```diff
@@ -7062,11 +7226,18 @@ class AsyncConverter : private SourceEntityWalker {
   }
 
   void addNewScope(const llvm::DenseSet<const Decl *> &Decls) {
-    ScopedNames.push_back({});
+    if (Scopes.empty()) {
+      Scopes.emplace_back(/*ContinuationName=*/Identifier());
+    } else {
+      // If the parent scope is nested in a continuation, the new one is also.
+      // Carry over the continuation name.
+      Identifier PreviousContinuationName = Scopes.back().ContinuationName;
+      Scopes.emplace_back(PreviousContinuationName);
+    }
     for (auto DeclAndNumRefs : Decls) {
```

Comment:
While you're here would you mind fixing this up to just "D" or something? I assume it was a map of Decl -> number of references at some point?